app/controllers/api/generic_works_controller.rb

```diff
@@ -0,0 +1,92 @@
+module API
+  # Adds an endpoint that customizes the update action on nested works
+  class GenericWorksController < ApplicationController
+    before_action :my_load_and_authorize_resource, only: [:update, :destroy, :show, :add_parent, :remove_parent]
```

Comment:
why are you bothering with the `only: ...` it looks like you've listed everything.